GrocifyApp.UI/src/pages/Settings.jsx

```diff
@@ -61,18 +61,53 @@ function Settings(props) {
     ];
 
     const [searchInput, setSearchInput] = useState('');
-    const [darkMode, setDarkMode] = useState(false);
+    const [darkMode, setDarkMode] = useState(props.userAuth ? props.userAuth.isDarkMode : false);
+    const [errorMessage, setErrorMessage] = useState('');
     const navigate = useNavigate();
 
+    const updateUserDarkMode = async () => {
+        
+        try {
+            var p = props.userAuth.password; //change
```

Comment:
User is a special entity, so we will not use the Update endpoint from Generic. Please create a new enpoint: "ToggleDarkMode". You don't need to pass information from user. You just need to pass the token in Header, and in the API you will have access to the authenticated user.

Reply:
To pass Token:

if (localStorage.getItem('token') == undefined) {
return;
}

const response = await fetch(api/User/ToggleDarkMode, {
method: 'PUT',
headers: {
'Content-Type': 'application/json',
'Authorization': Bearer ${localStorage.getItem('token')},
},
});

Reply:
if you followed my suggestion of saving darkMode in localStorage, don't forget to update it's value here